bin/fuel-core/src/cli/run.rs

```diff
@@ -577,3 +612,38 @@ fn start_pyroscope_agent(
         })
         .transpose()
 }
+
+#[cfg(feature = "rocksdb")]
+use fuel_core::{
+    database::database_description::DatabaseDescription,
+    state::historical_rocksdb::HistoricalRocksDB,
+};
+
+#[cfg(feature = "rocksdb")]
+use std::sync::Arc;
+
+#[cfg(feature = "rocksdb")]
+pub fn migrate_db<Description>(rocks_db: Arc<HistoricalRocksDB<Description>>)
+where
+    Description: DatabaseDescription,
+{
+    let entries = rocks_db.v1_entries();
+    for entry in entries {
+        match entry {
+            Err(e) => {
+                tracing::error!("Cannot read from database while migrating modification history: {e:?}");
+                break;
+            }
+            Ok((height, _change)) => {
+                // We ignore the change and read it again in a transaction, to ensure that we avoid
+                // fetching a stale change while a rollback to `height` is in progress.
+                while let Err(e) =
+                    rocks_db.migrate_modifications_history_at_height(height)
+                {
+                    tracing::warn!("Could not migrate modifications history at height {height}: {e:?}. Trying again.")
+                }
+            }
+        }
+        rocks_db.complete_migration();
```

Comment:
In case of `Err` we will just warn in log and continue with next entry, and then call `complete_migration();`.

I'm wondering if this leaves some entries not migrated? And if this is acceptable outcome.

I may be lacking some context, though.

Reply:
That's a good question. I also lack context to answer, probably @xgreenx knows

Reply:
Ah, I missed the point of your question. The rationale behind the loop is because I don't know whether we use pessmiistic or optimistic concurrency control, e.g. whether the migration transaction aborts in case of conflict. In the latter case, we need to retry the transaction.

Reply:
I think we need to skip failed entry and continue. But with changes that I asked from https://github.com/FuelLabs/fuel-core/pull/2233#discussion_r1782293032, you will not have cases where error can occur.